Firestore/Example/Tests/Local/FSTQueryCacheTests.mm

```diff
@@ -56,242 +56,267 @@ - (BOOL)isTestBaseClass {
 - (void)testReadQueryNotInCache {
   if ([self isTestBaseClass]) return;
 
+  FSTWriteGroup *group = [self.persistence startGroupWithAction:@"ReadQueryNotInCache"];
   XCTAssertNil([self.queryCache queryDataForQuery:_queryRooms]);
+  [self.persistence commitGroup:group];
 }
 
 - (void)testSetAndReadAQuery {
   if ([self isTestBaseClass]) return;
 
+  FSTWriteGroup *group = [self.persistence startGroupWithAction:@"SetAndReadQuery"];
   FSTQueryData *queryData = [self queryDataWithQuery:_queryRooms];
-  [self addQueryData:queryData];
+  [self.queryCache addQueryData:queryData group:group];
```

Comment:
This seems like a step in the wrong direction, since we're trying to make the transaction implicit in the `persistence` instance. I would have thought this change would remove the groups from the query cache's API and just have all writes go through `persistence.current_transaction`.

Reply:
This particular change isn't really a step forward or backwards, it's just removing the helper method `addQueryData` that created the group, used it, then committed it. That being said, I *think* I could update the `FSTQueryCache` API at this point to remove groups. It was helpful to not do that previously, since the goal right now is to identify all of the locations transactions are needed, which is a superset of all the places a group is currently used.

Reply:
I left `FSTQueryCache` as-is for now. I think the right time to change it is in the `FSTWriteGrou` -> `LevelDbTransaction` swap.

Reply:
Ok.

I think the issue I hadn't quite squared in my mind is that previously the test relied on getting the committed state right so that the next part of the test could proceed without needing the current transaction to have any interesting values. With this change in place, most tests can now just start a single transaction and validate within it without necessarily even committing (though we'll likely want them to).